jax/_src/numpy/fft.py

```diff
@@ -252,30 +258,35 @@ def fftfreq(n: int, d: ArrayLike = 1.0, *, dtype=None) -> Array:
           "The d argument of jax.numpy.fft.fftfreq only takes a single value. "
           "Got d = %s." % list(d))
 
-  k = jnp.zeros(n, dtype=dtype)
+  k = jnp.zeros(n, dtype=dtype, device=device)
   if n % 2 == 0:
     # k[0: n // 2 - 1] = jnp.arange(0, n // 2 - 1)
-    k = k.at[0: n // 2].set( jnp.arange(0, n // 2, dtype=dtype))
+    k = k.at[0: n // 2].set( jnp.arange(0, n // 2, dtype=dtype, device=device))
```

Comment:
I don't think `device` is needed here as it's a temporary array. Also if `device` is a sharding that is only valid for length-n, this will lead to an error.

Same below.